src/conventional/amdar_bufr2ioda.py

```diff
@@ -0,0 +1,508 @@
+        varAttrs[iodavar, qcName]['units'] = 'unitless'
+        n += 1
+
+    # Set units of the MetaData variables and all _FillValues.
+    for key in meta_keys:
+        varAttrs[(key, metaDataName)]['units'] = locationKeyList[meta_keys.index(key)][2]
+        if (locationKeyList[meta_keys.index(key)][1] == "string"):
+            varAttrs[(key, metaDataName)]['_FillValue'] = string_missing_value
+            obs_data[(key, metaDataName)] = np.array(data[key], dtype=object)
+        elif (locationKeyList[meta_keys.index(key)][1] == "integer"):
+            varAttrs[(key, metaDataName)]['_FillValue'] = int_missing_value
+            obs_data[(key, metaDataName)] = np.array(data[key], dtype=np.int32)
+        elif (locationKeyList[meta_keys.index(key)][1] == "long"):
+            varAttrs[(key, metaDataName)]['_FillValue'] = long_missing_value
+            obs_data[(key, metaDataName)] = np.array(data[key], dtype=np.int64)
+        elif (locationKeyList[meta_keys.index(key)][1] == "float"):
+            varAttrs[(key, metaDataName)]['_FillValue'] = float_missing_value
+            obs_data[(key, metaDataName)] = np.array(data[key], dtype=np.float32)
+        elif (locationKeyList[meta_keys.index(key)][1] == "double"):
+            varAttrs[(key, metaDataName)]['_FillValue'] = double_missing_value
+            obs_data[(key, metaDataName)] = np.array(data[key], dtype=np.float64)
+
+    # Transfer from the 1-D data vectors and ensure output data (obs_data) types using numpy.
+    iodavar = 'air_temperature'
+    obs_data[(iodavar, obsValName)] = np.array(data[iodavar], dtype=np.float32)
+    obs_data[(iodavar, obsErrName)] = np.full(nlocs, 1.2, dtype=np.float32)
+    obs_data[(iodavar, qcName)] = np.full(nlocs, 2, dtype=np.int32)
+    iodavar = 'specific_humidity'
+    obs_data[(iodavar, obsValName)] = np.array(data[iodavar], dtype=np.float32)
+    obs_data[(iodavar, obsErrName)] = np.full(nlocs, 0.75E-3, dtype=np.float32)
+    obs_data[(iodavar, qcName)] = np.full(nlocs, 2, dtype=np.int32)
+    iodavar = 'eastward_wind'
+    obs_data[(iodavar, obsValName)] = np.array(data[iodavar], dtype=np.float32)
+    obs_data[(iodavar, obsErrName)] = np.full(nlocs, 1.7, dtype=np.float32)
+    obs_data[(iodavar, qcName)] = np.full(nlocs, 2, dtype=np.int32)
+    iodavar = 'northward_wind'
+    obs_data[(iodavar, obsValName)] = np.array(data[iodavar], dtype=np.float32)
+    obs_data[(iodavar, obsErrName)] = np.full(nlocs, 1.7, dtype=np.float32)
+    obs_data[(iodavar, qcName)] = np.full(nlocs, 2, dtype=np.int32)
+
+    logging.debug("INFO: Writing file: " + output_file)
+
+    # setup the IODA writer
+    writer = iconv.IodaWriter(output_file, locationKeyList, DimDict)
+
+    # write everything out
+    writer.BuildIoda(obs_data, VarDims, varAttrs, AttrData)
+
+    logging.info("--- {:9.4f} total seconds ---".format(time.time() - start_time))
+
+
+# Replace BUFR missing value indicators with IODA missings.
+def assign_values(data, key):
```

Comment:
similar suggestion to above to reduce ifs here

Reply:
In this particular method, I must use the IF-testing for data type, because the outgoing data type can sometimes change from the input type.  The example use-case I found is the "height" variable in aircraft-AMDAR data is coming into BUFR as an integer, but we must have it as a float for use in UFO/JEDI.  Otherwise, I followed your advice in the other places.